### ISSUE TITLE
Speeding up Literate.jl workflow

### ISSUE THREAD

**Issue by jpfairbanks** [AlgebraicJulia/Catlab.jl#564]
Would it make sense to have a script in the doc folder for building one specific literate file and previewing it? I find that waiting for the docs to build really reduces the amount of iterations I am willing to go through in polishing the formatting of the literate files.

I know @TheCedarPrince is interested in revamping our docs infrastructure so maybe this is eliminated by a switch to Weave.jl or something.

**Comment by TheCedarPrince** [comment]
Hi @jpfairbanks - sounds like the Weave.jl pipeline I proposed could work.
Can you tell me more about what you are trying to do and ideally what you would want to see in the docs?
Just trying to get a feel for the pain points such that I could address them in some way.

**Comment by jpfairbanks** [comment]
Yeah, see my latest PRs #563 and #566. I am adding one file of docs and expect that everything else is going to be fine. My current workflow is to comment out most of the docs make.jl to get it to render the docs faster. 

In particular I have found disabling Literate.jl notebook execution to really accelerate it. 

```julia

  for (root, dirs, files) in walkdir(literate_dir)
    out_dir = joinpath(generated_dir, relpath(root, literate_dir))
    for file in files
      if last(splitext(file)) == ".jl"
        Literate.markdown(joinpath(root, file), out_dir;
                          config=config, documenter=true, credit=false)
        Literate.notebook(joinpath(root, file), out_dir;
                          execute=false, documenter=true, credit=false)
      end
    end
  end
```

Setting that execute to false makes the docs build much faster. But I would ideally have a function that can pick a single file in the literate folder, and render it to markdown and html as fast as possible.
